src/AppInstallerRepositoryCore/Microsoft/SQLiteIndex.h

```diff
@@ -146,6 +146,12 @@ namespace AppInstaller::Repository::Microsoft
         // Largely a utility function; should not be used to do work on behalf of the index by the caller.
         Utility::NormalizedName NormalizeName(std::string_view name, std::string_view publisher) const;
 
+        // Get all the dependencies for a specific manifest.
+        std::map<Manifest::Dependency, SQLite::rowid_t> GetDependenciesByManifestRowId(SQLite::rowid_t manifestRowId) const;
+        std::vector<std::pair<Manifest::Manifest, Utility::Version>> GetDependenciesByPackageId(AppInstaller::Manifest::string_t packageId) const;
+
+        // Get the latest version of a package.
+        std::optional<std::pair<SQLite::rowid_t, Utility::Version>> GetPackageLatestVersion(Manifest::string_t packageId, std::set<Utility::Version> versionsToExclude = {}) const;

```

Comment:
This is not needed.  A combination of `GetVersionKeysById` and `GetManifestIdByKey` already meet this need.